### PR TITLE
AG-11668 Remove Bean, Autowire, Optional decorators

### PR DESCRIPTION
**Pull request by StephenCooper**

### CODE REVIEW

File: community-modules/core/src/context/context.ts [hunk 1]
```diff
@@ -1,71 +1,162 @@
...
         if (!params || !params.beanClasses) {
             return;
         }
 
-        this.contextParams = params;
+        this.gridId = params.gridId;
+        this.beans.context = this;
 
-        this.logger = logger;
-        this.logger.log('>> creating ag-Application Context');
```
Comment (StephenCooper) [inline community-modules/core/src/context/context.ts:56]: Can we keep the logging in context?

File: community-modules/core/src/context/context.ts [hunk 2]
```diff
@@ -253,125 +197,34 @@ export class Context {
...
+        ModuleRegistry.__unRegisterGridModules(this.gridId);
     }
 
-    public destroyBean<T extends BaseBean | null | undefined>(bean: T): undefined {
-        if (!bean) {
-            return;
-        }
-
-        this.destroyBeans([bean]);
+    public destroyBean(bean: BaseBean | null | undefined): void {
```
Comment (StephenCooper) [inline community-modules/core/src/context/context.ts:209]: Not sure about this change. It was nice being able to destroy and cleanup in a single line. See that some methods now have two steps to destroy the bean and then assign null. What's the motivation for this?

File: community-modules/core/src/ctrlsService.ts [hunk 3]
```diff
@@ -47,9 +47,8 @@ interface ReadyParams {
 
 type CtrlType = keyof ReadyParams;
 
-@Bean(CtrlsService.NAME)
 export class CtrlsService extends BeanStub {
-    public static readonly NAME = 'ctrlsService';
```
Comment (StephenCooper) [inline community-modules/core/src/ctrlsService.ts:52]: What was this NAME property? Guessing not used for something else?

File: community-modules/core/src/filter/provided/date/dateCompWrapper.ts [hunk 4]
```diff
@@ -61,7 +61,8 @@ export class DateCompWrapper {
 
     public destroy(): void {
         this.alive = false;
-        this.dateComp = this.context.destroyBean(this.dateComp);
+        this.context.destroyBean(this.dateComp);
+        this.dateComp = undefined;
```
Comment (StephenCooper) [inline community-modules/core/src/filter/provided/date/dateCompWrapper.ts:65]: This pattern seems less effecient.

File: community-modules/core/src/headerRendering/rowContainer/headerRowContainerCtrl.ts [hunk 5]
```diff
@@ -97,7 +107,8 @@ export class HeaderRowContainerCtrl extends BeanStub {
         const refreshColumnGroups = () => {
             const groupRowCount = this.columnModel.getHeaderRowCount() - 1;
 
-            this.groupsRowCtrls = this.destroyBeans(this.groupsRowCtrls);
+            this.destroyBeans(this.groupsRowCtrls);
```
Comment (StephenCooper) [inline community-modules/core/src/headerRendering/rowContainer/headerRowContainerCtrl.ts:110]: Same thing here.

File: community-modules/core/src/rendering/cell/cellComp.ts [hunk 6]
```diff
@@ -296,7 +305,8 @@ export class CellComp extends Component implements TooltipParentComp {
 
     private destroyRenderer(): void {
         const { context } = this.beans;
-        this.cellRenderer = context.destroyBean(this.cellRenderer);
+        this.cellRenderer = undefined;
```
Comment (StephenCooper) [inline community-modules/core/src/rendering/cell/cellComp.ts:308]: These are the wrong way round.

File: community-modules/core/src/ctrlsService.ts (hunk 3 above)
Comment (AndrewGlazier) [reply community-modules/core/src/ctrlsService.ts:52]: This was used in several places like the `agGridReactUi` comp as a form of type safety for getting the bean name, I added type safety to `getBean` so it was no longer required.

**Pull request merged**